### ISSUE TITLE
the validate hm loss  is U-shape 

### ISSUE THREAD

**Issue by VCBE123** [xingyizhou/CenterNet#148]
I trained the ctdet-dla-34 on my dataset which contains nine categories with 9000 images.
The loss curve is below:
![Screenshot_2019-06-18 TensorBoard(3)](https://user-images.githubusercontent.com/30370620/59647460-d75bd400-91ad-11e9-85df-4cb0c9097710.png)
![Screenshot_2019-06-18 TensorBoard(2)](https://user-images.githubusercontent.com/30370620/59647463-db87f180-91ad-11e9-9c51-c17c58d518ca.png)
![Screenshot_2019-06-18 TensorBoard(1)](https://user-images.githubusercontent.com/30370620/59647465-dfb40f00-91ad-11e9-8a0a-a39916a0fac4.png)
![Screenshot_2019-06-18 TensorBoard](https://user-images.githubusercontent.com/30370620/59647472-e3479600-91ad-11e9-8624-eefcfeb2fd3a.png)

So why the val_hm_loss  is U-shape.  Is there any suggestion for this problem?

**Comment by xingyizhou** [comment]
Hi,
  We also observed this. However, the loss does not correspond to the final evaluation metric --- AP. The last model gives better AP than the lowest-loss model. We haven't observed overfitting on COCO if full data-augmentation is used.

**Comment by VCBE123** [comment]
I tried to adjust the weight of losses. If i set the weight of  loss_WH to 1, the loss curve looks good, but the performance dropped a lot. I also found the confidence score of the classification  is much lower than YOLO3.  So is the training realy overfitting ?

**Comment by xingyizhou** [comment]
Thanks for the report. We also observed the same thing. In my experiments, training longer always gives better performance on COCO. Not sure about the confidence score things.

**Comment by leemengxing** [comment]
@VCBE123 How to solve this problem? I tried torch.backends.cudnn.benchmark = False and it didn't work

**Comment by sisrfeng** [comment]
>  However, the loss does not correspond to the final evaluation metric --- AP. The last model gives better AP than the lowest-loss model.
> In my experiments, training longer always gives better performance on COCO

I thought the model with lowest val loss will get highest AP on val_set and test_set (no matter on coco, kitti or costum dataset) before seeing this.
If training longer always gives better performance, we don't need a val set and should train the model for a very long time. Is it really reasonable？

**Comment by fabiopoiesi** [comment]
The increasing trend of the validation loss is expected and is due to how the loss is defined.
Since the loss is designed to make the network to predict a peak, over time the network tend to predict gaussians with variances that tend to zero. Therefore at the beginning of the training making an error on the position is not as bad as making it at the end of the training. Basically, if the predicted point is not corresponding with the point of the ground truth (positive), the penalty for this is larger (high loss) at the end of the training.

Epoch 5 -> loss = 0.2789 -> peak conf = 0.4273
![Screenshot from 2021-09-08 18-03-21](https://user-images.githubusercontent.com/17005352/132544642-0f673322-be22-4ca4-b954-e894b192272a.png)

Epoch 20 -> loss = 8.3402  -> peak conf = 0.9791
![Screenshot from 2021-09-08 18-04-40](https://user-images.githubusercontent.com/17005352/132544892-0030fbeb-8a2c-4146-9c8e-d8c1237f9687.png)
